app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt

```diff
@@ -272,4 +286,20 @@ class PlaylistViewModel(
     }
     repository.loadPlaylistCover(playlist, onBitmapLoaded)
   }
+
+  fun preparePlaylistCoverForSpotify(): String? {
+    return try {
+      // Decode Base64 to Bitmap
+      val bitmap = base64ToBitmap(selectedPlaylist.value?.playlistCover!!)
+
+      // Re-encode the Bitmap to Base64 JPEG
+      val outputStream = ByteArrayOutputStream()
```

Comment:
You should close your flow when you don't use it anymore. Consider using a flow.use{}.

Reply:
This is an idea for it:
```
val byteArray = ByteArrayOutputStream().use {
        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, it)
        it.toByteArray()
      }
```